src/algebra/ordered_monoid.lean

```diff
@@ -47,6 +47,11 @@ class ordered_add_comm_monoid (α : Type*) extends add_comm_monoid α, partial_o
 (add_le_add_left       : ∀ a b : α, a ≤ b → ∀ c : α, c + a ≤ c + b)
 (lt_of_add_lt_add_left : ∀ a b c : α, a + b < a + c → b < c)
 
+/-- an `ordered_comm_monoid` with one-sided 'division'
+in the sense that if `a ≤ b`, there is some `c` for which `a * c = b` -/
+class has_exists_mul_of_le (α : Type u) [ordered_comm_monoid α] :=
+(exists_add_of_le : ∀ (a b : α), a ≤ b → ∃ (c : α), b = a * c)
+
 attribute [to_additive] ordered_comm_monoid
```

Comment:
These lines should probably be in the other order, to keep the `attribute` line next to the thing it applies to